dora/core/common/src/main/java/alluxio/membership/EtcdMembershipManager.java

```diff
@@ -93,16 +94,25 @@ public void join(WorkerInfo workerInfo) throws IOException {
     if (existingEntityBytes != null) {
       // It's not me, or not the same me.
       if (!Arrays.equals(existingEntityBytes, serializedEntity)) {
-        // In k8s this might be bcos worker pod restarting with the same worker identity
-        // but certain fields such as hostname has been changed. Register to ring path anyway.
-        WorkerServiceEntity existingEntity = new WorkerServiceEntity();
-        existingEntity.deserialize(existingEntityBytes);
-        LOG.warn("Same worker entity found bearing same workerid:{},"
-            + "existing WorkerServiceEntity to be overwritten:{},"
-            + "maybe benign if pod restart in k8s env or same worker"
-            + " scheduled to restart on another machine in baremetal env.",
-            workerInfo.getIdentity().toString(), existingEntity);
-        mAlluxioEtcdClient.createForPath(pathOnRing, Optional.of(serializedEntity));
+        if (mConf.isSet(PropertyKey.WORKER_IN_K8S_ENV)
+            && mConf.getBoolean(PropertyKey.WORKER_IN_K8S_ENV)) {
+          // In k8s this might be bcos worker pod restarting with the same worker identity
+          // but certain fields such as hostname has been changed. Register to ring path anyway.
```

Comment:
how is worker ID generated in a Kubernetes deployment? Is there a chance there are actually two *live* workers with the same ID but different hostnames?

Reply:
There's no special mechanism to generate IDs in k8s, so it's the random string, so theoretically that shouldn't happen.

Reply:
> no special mechanism to generate IDs in k8s, so it's the random string

Do you mean that the ID is not explicitly set via any Alluxio configuration properties by the operator or the helm chart, but gets generated at the first time the worker instance is created?

If so, sounds good to me as people rely on the k8s operator or the helm chart to configure new worker instances, rathar than copying Alluxio configs between workers.

Reply:
> If so, sounds good to me as people rely on the k8s operator or the helm chart to configure new worker instances, rathar than copying Alluxio configs between workers.

exactly, after ur worker id changes cluster deployment needs to rely on coordinated workflow. since theres no such thing for bare-metal, we have to safe-guard everything from code point of view.